frontend/amundsen_application/static/js/pages/LineagePage/index.tsx

```diff
@@ -0,0 +1,247 @@
+}
+export interface DispatchFromProps {
+  // getTableData: (
+  //   key: string,
+  //   searchIndex?: string,
+  //   source?: string
+  // ) => GetTableDataRequest;
+  getTableLineageDispatch: (key: string) => GetTableLineageRequest;
+  // openRequestDescriptionDialog: (
+  //   requestMetadataType: RequestMetadataType,
+  //   columnName: string
+  // ) => OpenRequestAction;
+  // searchSchema: (schemaText: string) => UpdateSearchStateRequest;
+}
+
+export interface MatchProps {
+  cluster: string;
+  database: string;
+  schema: string;
+  table: string;
+}
+
+export type LineageProps = PropsFromState &
+  DispatchFromProps &
+  RouteComponentProps<MatchProps>;
+
+const ErrorMessage = () => (
+  <div className="container error-label">
+    <Breadcrumb />
+    <label>{Constants.ERROR_MESSAGE}</label>
+  </div>
+);
+
+export class LineagePage extends React.Component<
+  LineageProps & RouteComponentProps<any>
+> {
+  private key: string;
+
+  state: any = {};
+
+  private didComponentMount: boolean = false;
+
+  componentDidMount() {
+    const { getTableLineageDispatch } = this.props;
+    this.key = this.getTableKey();
+    getTableLineageDispatch(this.key);
+
+    // // Centering the graph
+    // if (this.treeContainerRef.current) {
+    //   const dimensions = this.treeContainerRef.current!.getBoundingClientRect();
+    //   this.setState({
+    //     treeTranslate: {
+    //       x: dimensions.width / 2,
+    //       y: dimensions.height / 2,
+    //     },
+    //   });
+    // }
+    this.didComponentMount = true;
+  }
+
+  componentDidUpdate() {
+    const { getTableLineageDispatch } = this.props;
+    const newKey = this.getTableKey();
+    if (this.key !== newKey) {
+      this.key = newKey;
+      getTableLineageDispatch(this.key);
+    }
+  }
+
+  getDisplayName() {
+    const { match } = this.props;
+    const { params } = match;
+
+    return `Lineage Information | ${params.schema}.${params.table}`;
+  }
+
+  // ToDo: Should be moved to a common place
+  getTableKey() {
+    /*
+    This 'key' is the `table_uri` format described in metadataservice. Because it contains the '/' character,
+    we can't pass it as a single URL parameter without encodeURIComponent which makes ugly URLs.
+    DO NOT CHANGE
+    */
+    const { match } = this.props;
+    const { params } = match;
+
+    return `${params.database}://${params.cluster}.${params.schema}/${params.table}`;
+  }
+
+  handleClick = (e) => {
+    console.log(e);
+  };
+
+  render() {
+    // const { isLoading, statusCode, tableData } = this.props;
+    const { match, tableLineage, statusCode, isLoading } = this.props;
+    console.log('tableLineage', tableLineage);
+    console.log('statusCode', statusCode);
+
+    const { params } = match;
+    let innerContent;
+
+    // We want to avoid rendering the previous table's metadata before new data is fetched in componentDidMount
+    if (isLoading || !this.didComponentMount) {
+      innerContent = <LoadingSpinner />;
```

Comment:
We should try to avoid using the spinner. We are working on using the shimmer loader all over the place.